internal/controller/daemonsetpool_controller.go

```diff
@@ -240,19 +275,15 @@ func (r *DaemonsetPoolReconciler) updateObservedGeneration(ctx context.Context,
 }
 
 // get the Daemonset in the cluster with owner reference to the DaemonsetPool
-func (r *DaemonsetPoolReconciler) getDaemonset(ctx context.Context, daemonsetPool *kwoksigsv1beta1.DaemonsetPool) (*appsv1.DaemonSet, error) {
-	daemonset := &appsv1.DaemonSet{}
-	err := r.Get(ctx, client.ObjectKey{
-		Namespace: daemonsetPool.Namespace,
-		Name:      daemonsetPool.Name,
-	}, daemonset)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil, nil
-		}
+func (r *DaemonsetPoolReconciler) getDaemonset(ctx context.Context, daemonsetPool *kwoksigsv1beta1.DaemonsetPool) ([]appsv1.DaemonSet, error) {
+	daemonset := &appsv1.DaemonSetList{}
+	err := r.List(ctx, daemonset, client.InNamespace(daemonsetPool.Namespace), client.MatchingLabels{controllerLabel: daemonsetPool.Name})
+	if err != nil && strings.Contains(err.Error(), "does not exist") {
```

Comment:
you can use the following instead: errors.IsNotFound(err)
https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound